### ISSUE TITLE
Permissions from discord not synced with users.

### ISSUE THREAD

**Issue by djay** [mozilla/hubs#3710]
**Description**
With a room created from discord I can't seem to setup any permissions or roles on discord that affect users that enter into the hubs room. Only one user (who created the room, myself) seems to have any extra abilities.
Ideally I'm trying to give just speakers the ability to share their screen.

**To Reproduce**
1. create a room via the discord bot
2. create a speaker role in discord. Give this role "manage this channel" and "manage permissions" permissions for the channel with the room
3. create a new user in discord
3. give this user speaker role in discord
4. enter the hubs room
5. new user is not allowed to edit room settings, kick users or create objects

**Expected behavior**
5. new user is not allowed to edit room settings, kick users or create objects

**Screenshots**

**Hardware** 
 - Device:macbook
 - OS: macos
 - Browser: Chrome

**Additional context**
It would be better to give certain users permissions over specific objects to present

**Comment by brianpeiris** [comment]
You'll need to give those users the "kick members" permission in discord. I'll explain why:
The discord integration essentially gives you three levels of access to a room. The first level is basic room access, being able to enter the room in the first place -- as long as you can access the discord channel, you can enter rooms bound to it. The second level is room moderation, this gives you the ability to kick and mute users, but it also gives you the ability to manipulate objects in the room, including creating new objects -- this level corresponds to the "kick members" permission in discord. The third level is room modification, this gives you the ability to modify the room itself, including the ability to change room settings, change the scene, and close the room -- this level corresponds to the "manage channels" permission in discord.

**Comment by djay** [comment]
Ah that makes sense as I didn't try that. 
Documentation needs to be updated about the levels and also for creating objects. I'll submit a pr.

**Comment by brianpeiris** [comment]
Yup, our documentation didn't cover this in enough detail. Thanks for offering to fix it!

**Comment by djay** [comment]
ok. I tried it out but didn't seem to work

<img width="981" alt="Screen Shot 2021-01-19 at 10 25 21 am" src="https://user-images.githubusercontent.com/41700/104984316-149ade80-5a41-11eb-9674-ff86b04e5e4f.png">

I gave my speaker role kick members permission but when this member with speaker roles enters the room they don't get the to be able to create objects @brianpeiris

**Comment by djay** [comment]
@brianpeiris I created a PR on the readme even though I haven't got it to work yet but made it WIP
https://github.com/MozillaReality/hubs-discord-bot/pull/113

This readme is however hard to find so it would be better spelled out on the main discord bot page which I'm not sure where it is in github.

**Comment by brianpeiris** [comment]
Hmm. The only other issue I can think of is if the user with the "speaker" role is using a discord account that does not have a verified email address. We have a separate path in our code that allows those users access to the room, but does not give them any extra permissions.

Thanks for the PR. The docs for the discord bot are in the hubs-docs repo, if you're interested in adding to those as well: https://github.com/MozillaReality/hubs-docs/blob/master/docs/hubs-discord-bot.md

**Comment by djay** [comment]
@brianpeiris thats what it was. Thanks so much. I'll add a note about that in both places

**Comment by djay** [comment]
Done in https://github.com/MozillaReality/hubs-docs/pull/91.

**Comment by djay** [comment]
Not sure it needs to be spelled out in more detail but a table could be added similar to this

|  | Viewer | Moderator | Owner |
|-|:-:|:-:|:-:|
| Discord<br>"View Channel" | x | x | x |
| Discord<br>"Kick Members" |  | x | x |
| Discord<br>"Manage Channel" |  |  | x |
| Hubs<br>Enter Room | I | I | I |
| Hubs<br>Kick Member |  |  |  |
